### PR TITLE
Ensure all item updates set the UT for Cosmos data

### PR DESCRIPTION
**Pull request by alexfca**
In a previous PR the CT (CreateTime) and UT (UpdateTime) was added to all Cosmos docs and implemented for Upsert.
I forgot to also update the UT for explicit Update methods. 

This PR adds a "item.SetUpdateTime()" to all Update statements.
As well as this all private setXXX() methods have been refactored to use a shared UpdateDocument() method

### CODE REVIEW

File: appservice/storage/cosmosdb/appservice_events_table.go [hunk 1]
```diff
@@ -144,18 +144,6 @@ func getEvent(s *eventsStatements, ctx context.Context, pk string, docId string)
 	return &response, err
 }
 
-func setEvent(s *eventsStatements, ctx context.Context, event eventCosmosData) (*eventCosmosData, error) {
```
Comment (alexfca) [inline appservice/storage/cosmosdb/appservice_events_table.go:147]: Removal of private methods..

File: internal/cosmosdbapi/client.go [hunk 2]
```diff
@@ -141,6 +141,26 @@ func GetDocumentOrNil(connection CosmosConnection, config CosmosConfig, ctx cont
 	return nil
 }
 
+func UpdateDocument(ctx context.Context,
```
Comment (alexfca) [inline internal/cosmosdbapi/client.go:144]: This is the new shared method

File: federationsender/storage/cosmosdb/outbound_peeks_table.go [hunk 3]
```diff
@@ -206,20 +194,21 @@ func (s *outboundPeeksStatements) RenewOutboundPeek(
...
 	}
 
-	res.OutboundPeek.RenewedTimestamp = nowMilli
-	res.OutboundPeek.RenewalInterval = renewalInterval
+	item.SetUpdateTime()
+	item.OutboundPeek.RenewedTimestamp = nowMilli
+	item.OutboundPeek.RenewalInterval = renewalInterval
 
-	_, err = setOutboundPeek(s, ctx, *res)
+	_, err = cosmosdbapi.UpdateDocument(ctx, s.db.connection, s.db.cosmosConfig.DatabaseName, s.db.cosmosConfig.ContainerName, item.Pk, item.ETag, item.Id, item)
```
Comment (alexfca) [inline federationsender/storage/cosmosdb/outbound_peeks_table.go:211]: The use of the shared method

File: federationsender/storage/cosmosdb/outbound_peeks_table.go [hunk 4]
```diff
@@ -206,20 +194,21 @@ func (s *outboundPeeksStatements) RenewOutboundPeek(
...
 	}
 
-	if res == nil {
+	if item == nil {
 		return
 	}
 
-	res.OutboundPeek.RenewedTimestamp = nowMilli
-	res.OutboundPeek.RenewalInterval = renewalInterval
+	item.SetUpdateTime()
```
Comment (alexfca) [inline federationsender/storage/cosmosdb/outbound_peeks_table.go:207]: Setting the UT

**Review by jaredjn — APPROVED** [review]

**Pull request merged**